internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -218,6 +218,17 @@ func (a *Adapter) prepareSnapshotForPipelineRun(pipelineRun *tektonv1.PipelineRu
 		snapshot.Labels[gitops.BuildPipelineRunFinishTimeLabel] = strconv.FormatInt(time.Now().Unix(), 10)
 	}
 
+	if pipelineRun.Status.StartTime != nil {
```

Comment:
I would like to do some change for this code, because I notice [STONEINTG-998](https://issues.redhat.com/browse/STONEINTG-998) mentions `buildPLR controller can add metadata into build PLR - pr-group and pr-group-sha`, so I want to move these code to a new fucntion EnsurePipelineRunIsAnnotatedAndLabelled(), then these annotations/labels can be copied from pipelineRun to component snapshot by [CopySnapshotLabelsAndAnnotations()](https://github.com/konflux-ci/integration-service/blob/main/internal/controller/buildpipeline/buildpipeline_adapter.go#L212C9-L212C41) and used to search PR group pipelineRun and snasphot. HDYT?